### PR TITLE
docs: Add scheduler description.

### PR DESCRIPTION
**Pull request by ceache**

### CODE REVIEW

File: docs/source/scheduler.rst [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+The Treadmill scheduler is responsible for finding application placement on the
+Treadmill cell. Given the collection of servers and their attributes, such as
+available capacity and other constraints, Treadmill scheduler will place
+applications to satisfy application demands as described in its Treadmill
+manifest.
+
+Treadmill scheduler has the following high level features:
+
+ * It brokers placement decision between independent users. Treadmill scheduler
```
Comment (DovOps) [inline docs/source/scheduler.rst]: placement decision[s]?

File: docs/source/scheduler.rst [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+   is multi-tenant, and different users do not coordinate their application
+   demand.
+ * Supports capacity reservations and opportunistic (non-reserved) placement.
+ * Server presence aware, will reschedule applications if the server goes down.
+ * Server topology aware, can spread or stack applications across different
+   servers, racks and pods.
+ * Support placement constraints, reboot aware.
```
Comment (DovOps) [inline docs/source/scheduler.rst]: Support[s]

File: docs/source/scheduler.rst [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+When users submit application to run on a Treadmill cell, they specify the
+target partition. Partitions form a unit of administration. They can be
+restricted for subset of users, different partitions can have different alert
+thresholds, reboot schedules, or be comprised from servers that have different
+attributes that are not expressed in Treadmill manifest (such as a set of
+firewall rules, connectivity so downstream resources, very particular hardware,
+etc.)
+
+While there is no limitation on number of partitions, dividing cell into
```
Comment (DovOps) [inline docs/source/scheduler.rst]: This is confusing. You first mention Treadmill operates on a partition, and that when deploying to a cell you need to specify the target partition, and then you say dividing into partitions is an anti-pattern.  Are you saying that dividing into MANY partitions is an anti pattern? What is the recommended number of partitions? What is an example of multiple partitions? Can partitions be dynamically resized while TM is running? I think this could use some more clarity.

File: docs/source/scheduler.rst [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+adjusted anytime by the application owner.
+
+Priority allows individual users to express which applications are more
+important for them. Priority values range from 100 (most important) to 0.
+Priorities matter only within a given allocation.
+
+Advanced: Priority 0 has special meaning. Setting application priority to zero
+for instruct the scheduler to place this application at the very end of the
+global queue. As such, priority zero applications will run only when there is
+no capacity pressure on the cell.
```
Comment (DovOps) [inline docs/source/scheduler.rst]: --- They're probably also highly likely to be evicted -e .g. they're like "spot" compute jobs, of sorts, right?

File: docs/source/scheduler.rst (hunk 3 above)
Comment (ceache) [reply docs/source/scheduler.rst]: Rewrote the section

File: docs/source/scheduler.rst (hunk 2 above)
Comment (ceache) [reply docs/source/scheduler.rst]: fixed

File: docs/source/scheduler.rst (hunk 1 above)
Comment (ceache) [reply docs/source/scheduler.rst]: fixed

File: docs/source/scheduler.rst [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+Server topology constraints
+---------------------------
+
+Servers in the cell are organized in a hierarchy that aims to reflect physical
+topology of the datacenter.
+
+Typical structure is server, rack, pod (rack is collection of servers, pod is
+collection of racks).
+
+Treadmill scheduler supports two placement strategies - stack and spread. The
```
Comment (DovOps) [inline docs/source/scheduler.rst]: I recommend explaining stack and spread placement strategy a bit better. You kind of explain spread, but don't really explain stack.  I found this vague/confusing to read.

File: docs/source/scheduler.rst [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+
+.. _section-application-lifetime:
+
+Application lifetime constraints and reboot awareness
+-----------------------------------------------------
+
+Treadmill scheduler is aware of the expected lifetime (time till reboot) for
+servers in the cell. Typical reboot schedule can be once in three weeks,
+percentage of cell rebooting every day spread across 3 weeks.
```
Comment (DovOps) [inline docs/source/scheduler.rst]: I'd reword.   The typical reboot schedule of a server might be once every three weeks, and within a cell, there is often a fixed/known percentage of the cell's servers rebooting on a given day.  (Is that true?) - Current reading is somewhat vague.

File: docs/source/scheduler.rst (hunk 4 above)
Comment (ceache) [reply docs/source/scheduler.rst]: Moved content to priority-0 section below. Clarified that section

File: docs/source/scheduler.rst (hunk 6 above)
Comment (ceache) [reply docs/source/scheduler.rst]: reworded

File: docs/source/scheduler.rst (hunk 5 above)
Comment (ceache) [reply docs/source/scheduler.rst]: rewritten
